File: go-client-sdk/pkg/models/operations/patchsettingsintegrationsintegrationid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type PatchSettingsIntegrationsIntegrationIDPathParams struct {
	IntegrationID string `pathParam:"style=simple,explode=false,name=integrationId"`
}

type PatchSettingsIntegrationsIntegrationIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type PatchSettingsIntegrationsIntegrationIDRequest struct {
	PathParams PatchSettingsIntegrationsIntegrationIDPathParams
	Request    *shared.CodatPublicAPIModelsClientsIntegrationSettingsPatchModel `request:"mediaType=application/json"`
	Security   PatchSettingsIntegrationsIntegrationIDSecurity
}

type PatchSettingsIntegrationsIntegrationIDResponse struct {
	CodatPublicAPIModelsClientsIntegrationSettingsModel *shared.CodatPublicAPIModelsClientsIntegrationSettingsModel
	ContentType                                         string
	StatusCode                                          int
}

File: typescript-client-sdk/src/sdk/models/operations/patchsettingsintegrationsintegrationid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PatchSettingsIntegrationsIntegrationIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=integrationId" })
  integrationId: string;
}


export class PatchSettingsIntegrationsIntegrationIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class PatchSettingsIntegrationsIntegrationIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PatchSettingsIntegrationsIntegrationIdPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.CodatPublicApiModelsClientsIntegrationSettingsPatchModel;

  @SpeakeasyMetadata()
  security: PatchSettingsIntegrationsIntegrationIdSecurity;
}


export class PatchSettingsIntegrationsIntegrationIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsClientsIntegrationSettingsModel?: shared.CodatPublicApiModelsClientsIntegrationSettingsModel;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
